src/api/lib/backend/api/build_results/binaries.rb

```diff
@@ -31,6 +31,55 @@ def self.build_dependency_info(project, package, repository, architecture)
           path += "?package=#{CGI.escape(package)}&view=pkgnames"
           Backend::Connection.get(path).body.force_encoding("UTF-8")
         end
+
+        # Returns the available binaries for the project
+        def self.available_in_project(project)
+          path = "/build/#{CGI.escape(project)}/_availablebinaries"
+          transform_binary_packages_response(Backend::Connection.get(path).body.force_encoding("UTF-8"))
+        end
+
+        # Returns the available binaries for the urls given from the project
+        def self.available_in_urls(project, urls)
+          return {} if urls.empty?
+          path = "/build/#{CGI.escape(project)}/_availablebinaries"
+          query = []
+          urls.each do |url|
+            query << "url=#{CGI.escape(url)}"
+          end
+          path += "?#{query.join('&')}"
+          transform_binary_packages_response(Backend::Connection.get(path).body.force_encoding("UTF-8"))
+        end
+
+        # Returns the available binaries for the obs_repositories given from the project
+        def self.available_in_repositories(project, repositories)
+          return {} if repositories.empty?
+          path = "/build/#{CGI.escape(project)}/_availablebinaries"
+          query = []
+          repositories.each do |repository|
+            query << "path=#{CGI.escape(repository)}"
+          end
+          path += "?#{query.join('&')}"
+          transform_binary_packages_response(Backend::Connection.get(path).body.force_encoding("UTF-8"))
+        end
+
+        # Transforms the output of the available_in_repositories, available_in_urls and available_in_project methods to a hash containing
+        # the name of the binary as keys and the architectures as the value
+        def self.transform_binary_packages_response(response)
+          list = {}
+          parsed_response = Xmlhash.parse(response)
+          return list if parsed_response.blank?
+          packages = [parsed_response["packages"]].flatten
+          packages.each do |build|
+            architectures = [build["arch"]].flatten
+            packages = [build["name"]].flatten
+            packages.each do |package|
+              architectures = architectures.concat(list[package]) if list[package]
+              list[package] = architectures.uniq
+            end
+          end
+          list
```

Comment:
I find this method quite complicated. Maybe we can just store the package names for now and ignore the architectures which would already make it a lot simpler.

```ruby
def self.transform_binary_packages_response(response)
  result = []
  parsed_response = Xmlhash.parse(response)
  [parsed_response["packages"]].flatten.each do |group|
    result << group['name']
  end
  result.flatten.uniq
end
```

Reply:
Do you think that dropping the architectures info is better than having a "complex" method?

Reply:
Yes, as we don't use the architecture anyway! You know: Fuck the future, program for today :rofl: :smile:

Reply:
I wanted to have [arch1, arch2, ...] as part of the label for the dropdown that is shown.... may I drop information because the method is too complex at first sight?
😢

Reply:
Okay, if you think you need to keep it I'm fine with it! But we shouldn't encourage users to set the architecture anyway -> we should introduce an expert mode as well and leave nil as default-

Reply:
Just wanted to show the archs... not enforcing the user to use that...